applications/DeepSpeed-Chat/training/utils/perf.py

```diff
@@ -48,79 +41,126 @@ def print_throughput(hf_model, args, e2e_time, rank=0):
+
+        actor_model._num_params = sum([
             p.ds_numel if hasattr(p, "ds_tensor") else p.numel()
-            for p in hf_model.parameters()
+            for p in actor_model.parameters()
         ])
-        params_in_billions = hf_model._num_params / (1e9)
+        actor_params_in_billions = actor_model._num_params / (1e9)
+
+        critic_model._num_params = sum([
+            p.ds_numel if hasattr(p, "ds_tensor") else p.numel()
+            for p in critic_model.parameters()
+        ])
+        critic_params_in_billions = critic_model._num_params / (1e9)
 
         # Megatron paper's formula to calculate training flops
-        train_flops_per_iteration = (
-            24 * checkpoint_activations_factor * batch_size * seq_length *
-            num_layers *
-            (hidden_size**2)) * (1.0 + (seq_length / (6.0 * hidden_size)) +
-                                 (vocab_size /
-                                  (16.0 * num_layers * hidden_size)))
 
-        train_tflops = train_flops_per_iteration / (train_time *
-                                                    gpus_per_model * (10**12))
+        actor_train_flops_per_iteration = calculate_flops(
+            actor_checkpoint_activations_factor, batch_size, seq_length,
+            actor_hf_config)
+        critic_train_flops_per_iteration = calculate_flops(
+            critic_checkpoint_activations_factor, batch_size, seq_length,
+            critic_hf_config)
+
+        total_train_flops = actor_train_flops_per_iteration + critic_train_flops_per_iteration
+        train_tflops = total_train_flops / (train_time * gpus_per_model *
+                                            (10**12))
 
         gen_bs = args.per_device_generation_batch_size * gpus_per_model
 
-        # Modified formula for calculating flops in forward pass only
+        # Modified formula for calculating flops in the forward pass only
         gen_flops_per_iteration = (
-            24 * gen_bs * seq_length * num_layers *
-            (hidden_size**2)) * (1.0 + (seq_length / (6.0 * hidden_size)) +
-                                 (vocab_size /
-                                  (16.0 * num_layers * hidden_size)))
+            24 * gen_bs * seq_length * actor_num_layers *
+            (actor_hidden_size**2)) * (
+                1.0 + (seq_length / (6.0 * actor_hidden_size)) +
+                (actor_vocab_size /
+                 (16.0 * actor_num_layers * actor_hidden_size)))
 
         gen_tflops = gen_flops_per_iteration / (gen_exp_time * gpus_per_model *
                                                 (10**12))
 
-        if hf_config.torch_dtype == "float16":
+        if actor_hf_config.torch_dtype == torch.float16:
             num_bytes = 2
-        elif hf_config.torch_dtype == "float32":
+        elif actor_hf_config.torch_dtype == torch.float32:
             num_bytes = 4
         else:
-            num_bytes = 1
+            num_bytes = -1
```

Comment:
Clarification question: Why this is `-1`?

Reply:
To avoid a silent bug, see a few lines below where I check if its is -1, then print NA, instead of printing wrong bandwidth